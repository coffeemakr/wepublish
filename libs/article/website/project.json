{
  "name": "article-website",
  "$schema": "../../../node_modules/nx/schemas/project-schema.json",
  "sourceRoot": "libs/article/website/src",
  "projectType": "library",
  "tags": [],
  "targets": {
    "lint": {
      "executor": "@nx/linter:eslint",
      "outputs": ["{options.outputFile}"],
      "options": {
        "lintFilePatterns": ["libs/article/website/**/*.{ts,tsx,js,jsx}"]
      }
    },
    "build": {
      "executor": "@nx/js:tsc",
      "outputs": ["{options.outputPath}"],
      "options": {
        "outputPath": "dist/libs/article/website",
        "tsConfig": "libs/article/website/tsconfig.lib.json",
        "packageJson": "libs/article/website/package.json",
        "main": "libs/article/website/src/index.ts",
        "assets": ["libs/article/website/*.md"],
        "updateBuildableProjectDepsInPackageJson": true
      }
    },
    "test": {
      "executor": "@nx/jest:jest",
      "outputs": ["{workspaceRoot}/coverage/{projectRoot}"],
      "options": {
        "jestConfig": "libs/article/website/jest.config.ts",
        "passWithNoTests": true
      }
    }
  }
}
