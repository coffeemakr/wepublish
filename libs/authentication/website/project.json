{
  "name": "authentication-website",
  "$schema": "../../../node_modules/nx/schemas/project-schema.json",
  "sourceRoot": "libs/authentication/website/src",
  "projectType": "library",
  "tags": [],
  "targets": {
    "lint": {
      "executor": "@nx/linter:eslint",
      "outputs": ["{options.outputFile}"],
      "options": {
        "lintFilePatterns": ["libs/authentication/website/**/*.{ts,tsx,js,jsx}"]
      }
    },
    "build": {
      "executor": "@nx/js:tsc",
      "outputs": ["{options.outputPath}"],
      "options": {
        "outputPath": "dist/libs/authentication/website",
        "tsConfig": "libs/authentication/website/tsconfig.lib.json",
        "packageJson": "libs/authentication/website/package.json",
        "main": "libs/authentication/website/src/index.ts",
        "assets": ["libs/authentication/website/*.md"],
        "updateBuildableProjectDepsInPackageJson": true
      }
    },
    "test": {
      "executor": "@nx/jest:jest",
      "outputs": ["{workspaceRoot}/coverage/{projectRoot}"],
      "options": {
        "jestConfig": "libs/authentication/website/jest.config.ts",
        "passWithNoTests": true
      }
    }
  }
}
