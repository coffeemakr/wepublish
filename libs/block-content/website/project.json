{
  "name": "block-content-website",
  "$schema": "../../../node_modules/nx/schemas/project-schema.json",
  "sourceRoot": "libs/block-content/website/src",
  "projectType": "library",
  "tags": [],
  "targets": {
    "lint": {
      "executor": "@nx/linter:eslint",
      "outputs": ["{options.outputFile}"],
      "options": {
        "lintFilePatterns": ["libs/block-content/website/**/*.{ts,tsx,js,jsx}"]
      }
    },
    "build": {
      "executor": "@nx/js:tsc",
      "outputs": ["{options.outputPath}"],
      "options": {
        "outputPath": "dist/libs/block-content/website",
        "tsConfig": "libs/block-content/website/tsconfig.lib.json",
        "packageJson": "libs/block-content/website/package.json",
        "main": "libs/block-content/website/src/index.ts",
        "assets": ["libs/block-content/website/*.md"],
        "updateBuildableProjectDepsInPackageJson": true
      }
    },
    "test": {
      "executor": "@nx/jest:jest",
      "outputs": ["{workspaceRoot}/coverage/{projectRoot}"],
      "options": {
        "jestConfig": "libs/block-content/website/jest.config.ts",
        "passWithNoTests": true
      }
    }
  }
}
