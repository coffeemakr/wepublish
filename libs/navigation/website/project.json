{
  "name": "navigation-website",
  "$schema": "../../../node_modules/nx/schemas/project-schema.json",
  "sourceRoot": "libs/navigation/website/src",
  "projectType": "library",
  "tags": [],
  "targets": {
    "lint": {
      "executor": "@nx/linter:eslint",
      "outputs": ["{options.outputFile}"],
      "options": {
        "lintFilePatterns": ["libs/navigation/website/**/*.{ts,tsx,js,jsx}"]
      }
    },
    "build": {
      "executor": "@nx/js:tsc",
      "outputs": ["{options.outputPath}"],
      "options": {
        "outputPath": "dist/libs/navigation/website",
        "tsConfig": "libs/navigation/website/tsconfig.lib.json",
        "packageJson": "libs/navigation/website/package.json",
        "main": "libs/navigation/website/src/index.ts",
        "assets": ["libs/navigation/website/*.md"],
        "updateBuildableProjectDepsInPackageJson": true
      }
    },
    "test": {
      "executor": "@nx/jest:jest",
      "outputs": ["{workspaceRoot}/coverage/{projectRoot}"],
      "options": {
        "jestConfig": "libs/navigation/website/jest.config.ts",
        "passWithNoTests": true
      }
    }
  }
}
