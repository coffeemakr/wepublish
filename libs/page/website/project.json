{
  "name": "page-website",
  "$schema": "../../../node_modules/nx/schemas/project-schema.json",
  "sourceRoot": "libs/page/website/src",
  "projectType": "library",
  "tags": [],
  "targets": {
    "lint": {
      "executor": "@nx/linter:eslint",
      "outputs": ["{options.outputFile}"],
      "options": {
        "lintFilePatterns": ["libs/page/website/**/*.{ts,tsx,js,jsx}"]
      }
    },
    "build": {
      "executor": "@nx/js:tsc",
      "outputs": ["{options.outputPath}"],
      "options": {
        "outputPath": "dist/libs/page/website",
        "tsConfig": "libs/page/website/tsconfig.lib.json",
        "packageJson": "libs/page/website/package.json",
        "main": "libs/page/website/src/index.ts",
        "assets": ["libs/page/website/*.md"],
        "updateBuildableProjectDepsInPackageJson": true
      }
    },
    "test": {
      "executor": "@nx/jest:jest",
      "outputs": ["{workspaceRoot}/coverage/{projectRoot}"],
      "options": {
        "jestConfig": "libs/page/website/jest.config.ts",
        "passWithNoTests": true
      }
    }
  }
}
