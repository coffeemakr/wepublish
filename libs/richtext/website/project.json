{
  "name": "richtext-website",
  "$schema": "../../../node_modules/nx/schemas/project-schema.json",
  "sourceRoot": "libs/richtext/website/src",
  "projectType": "library",
  "tags": [],
  "targets": {
    "lint": {
      "executor": "@nx/linter:eslint",
      "outputs": ["{options.outputFile}"],
      "options": {
        "lintFilePatterns": ["libs/richtext/website/**/*.{ts,tsx,js,jsx}"]
      }
    },
    "build": {
      "executor": "@nx/js:tsc",
      "outputs": ["{options.outputPath}"],
      "options": {
        "outputPath": "dist/libs/richtext/website",
        "tsConfig": "libs/richtext/website/tsconfig.lib.json",
        "packageJson": "libs/richtext/website/package.json",
        "main": "libs/richtext/website/src/index.ts",
        "assets": ["libs/richtext/website/*.md"],
        "updateBuildableProjectDepsInPackageJson": true
      }
    },
    "test": {
      "executor": "@nx/jest:jest",
      "outputs": ["{workspaceRoot}/coverage/{projectRoot}"],
      "options": {
        "jestConfig": "libs/richtext/website/jest.config.ts",
        "passWithNoTests": true
      }
    }
  }
}
