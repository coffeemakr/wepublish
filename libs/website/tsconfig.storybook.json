{
  "extends": "./tsconfig.json",
  "compilerOptions": {
    "emitDecoratorMetadata": true,
    "outDir": ""
  },
  "files": [
    "../../node_modules/@nx/react/typings/styled-jsx.d.ts",
    "../../node_modules/@nx/react/typings/cssmodule.d.ts",
    "../../node_modules/@nx/react/typings/image.d.ts"
  ],
  "exclude": ["src/**/*.spec.ts", "src/**/*.spec.js", "src/**/*.spec.tsx", "src/**/*.spec.jsx"],
  "include": [
    "../**/website/src/**/*.stories.ts",
    "../**/website/src/**/*.stories.js",
    "../**/website/src/**/*.stories.jsx",
    "../**/website/src/**/*.stories.tsx",
    "../**/website/src/**/*.stories.mdx",
    ".storybook/*.js",
    "preview.tsx",
    "main.ts"
  ]
}
