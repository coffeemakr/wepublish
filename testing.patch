diff --git a/libs/api/src/lib/db/article.ts b/libs/api/src/lib/db/article.ts
index 7359af6a..ea27be09 100644
--- a/libs/api/src/lib/db/article.ts
+++ b/libs/api/src/lib/db/article.ts
@@ -14,11 +14,11 @@ export interface ArticleData {
   readonly lead?: string | null
   readonly seoTitle?: string | null
   readonly slug?: string | null
-  readonly tags: string[]
+  readonly tags?: string[]
 
   readonly canonicalUrl?: string | null
 
-  readonly properties: MetadataProperty[]
+  readonly properties?: MetadataProperty[]
 
   readonly imageID?: string | null
   readonly authors: ArticleRevisionAuthor[]
@@ -107,7 +107,7 @@ export type ArticleRevisionWithRelations = PrismaArticleRevision & {
 
 export type ArticleWithRevisions = PrismaArticle & {
   draft: ArticleRevisionWithRelations | null
-  pending: ArticleRevisionWithRelations | null
+  pending?: ArticleRevisionWithRelations | null
   published: ArticleRevisionWithRelations | null
 }
 
diff --git a/libs/api/src/lib/db/page.ts b/libs/api/src/lib/db/page.ts
index fb73c289..6f0ed1b5 100644
--- a/libs/api/src/lib/db/page.ts
+++ b/libs/api/src/lib/db/page.ts
@@ -14,9 +14,9 @@ export interface PageData {
 
   readonly title: string
   readonly description?: string | null
-  readonly tags: string[]
+  readonly tags?: string[]
 
-  readonly properties: MetadataProperty[]
+  readonly properties?: MetadataProperty[]
 
   readonly imageID?: string | null
 
@@ -85,7 +85,7 @@ export type PageRevisionWithProperties = PrismaPageRevision & {
 
 export type PageWithRevisions = PrismaPage & {
   draft: PageRevisionWithProperties | null
-  pending: PageRevisionWithProperties | null
+  pending?: PageRevisionWithProperties | null
   published: PageRevisionWithProperties | null
 }
 
diff --git a/libs/api/src/lib/graphql/article.ts b/libs/api/src/lib/graphql/article.ts
index 8d80fb27..77357e01 100644
--- a/libs/api/src/lib/graphql/article.ts
+++ b/libs/api/src/lib/graphql/article.ts
@@ -275,7 +275,7 @@ export const GraphQLPublicArticle: GraphQLObjectType<PublicArticle, Context> =
         type: GraphQLNonNull(GraphQLList(GraphQLNonNull(GraphQLMetadataPropertyPublic))),
         resolve: ({properties}) => {
           return properties
-            .filter(property => property.public)
+            ?.filter(property => property.public)
             .map(({key, value}) => ({key, value}))
         }
       },
diff --git a/libs/api/src/lib/graphql/article/article.private-mutation.ts b/libs/api/src/lib/graphql/article/article.private-mutation.ts
index 43a6a3ee..049c7e54 100644
--- a/libs/api/src/lib/graphql/article/article.private-mutation.ts
+++ b/libs/api/src/lib/graphql/article/article.private-mutation.ts
@@ -178,7 +178,8 @@ export const duplicateArticle = async (
       createMany: {
         data: socialMediaAuthors.map(({authorId}) => ({authorId}))
       }
-    }
+    },
+    blocks: []
   }
 
   return articleClient.create({
@@ -303,7 +304,8 @@ export const unpublishArticle = async (
               createMany: {
                 data: socialMediaAuthors.map(({authorId}) => ({authorId}))
               }
-            }
+            },
+            blocks: []
           },
           update: {
             ...revision,
@@ -333,7 +335,8 @@ export const unpublishArticle = async (
               createMany: {
                 data: socialMediaAuthors.map(({authorId}) => ({authorId}))
               }
-            }
+            },
+            blocks: []
           }
         }
       },
@@ -491,7 +494,8 @@ export const publishArticle = async (
                 createMany: {
                   data: socialMediaAuthors.map(({authorId}) => ({authorId}))
                 }
-              }
+              },
+              blocks: []
             },
             update: {
               ...revision,
@@ -521,7 +525,8 @@ export const publishArticle = async (
                 createMany: {
                   data: socialMediaAuthors.map(({authorId}) => ({authorId}))
                 }
-              }
+              },
+              blocks: []
             }
           }
         },
@@ -579,7 +584,8 @@ export const publishArticle = async (
               createMany: {
                 data: socialMediaAuthors.map(({authorId}) => ({authorId}))
               }
-            }
+            },
+            blocks: []
           },
           update: {
             ...revision,
@@ -609,7 +615,8 @@ export const publishArticle = async (
               createMany: {
                 data: socialMediaAuthors.map(({authorId}) => ({authorId}))
               }
-            }
+            },
+            blocks: []
           }
         }
       },
diff --git a/libs/api/src/lib/graphql/page.ts b/libs/api/src/lib/graphql/page.ts
index 56d380f0..6b63518e 100644
--- a/libs/api/src/lib/graphql/page.ts
+++ b/libs/api/src/lib/graphql/page.ts
@@ -210,7 +210,7 @@ export const GraphQLPublicPage = new GraphQLObjectType<PublicPage, Context>({
     properties: {
       type: GraphQLNonNull(GraphQLList(GraphQLNonNull(GraphQLMetadataPropertyPublic))),
       resolve: ({properties}) => {
-        return properties.filter(property => property.public).map(({key, value}) => ({key, value}))
+        return properties?.filter(property => property.public).map(({key, value}) => ({key, value}))
       }
     },
 
diff --git a/libs/api/src/lib/graphql/page/page.private-mutation.ts b/libs/api/src/lib/graphql/page/page.private-mutation.ts
index 8d21a995..456df84e 100644
--- a/libs/api/src/lib/graphql/page/page.private-mutation.ts
+++ b/libs/api/src/lib/graphql/page/page.private-mutation.ts
@@ -140,7 +140,8 @@ export const duplicatePage = async (
       createMany: {
         data: duplicatedProperties
       }
-    }
+    },
+    blocks: []
   }
 
   return pageClient.create({
@@ -236,7 +237,8 @@ export const unpublishPage = async (
               createMany: {
                 data: properties
               }
-            }
+            },
+            blocks: []
           },
           update: {
             ...revision,
@@ -250,7 +252,8 @@ export const unpublishPage = async (
               createMany: {
                 data: properties
               }
-            }
+            },
+            blocks: []
           }
         }
       },
@@ -387,7 +390,8 @@ export const publishPage = async (
                 createMany: {
                   data: properties
                 }
-              }
+              },
+              blocks: []
             },
             update: {
               ...revision,
@@ -401,7 +405,8 @@ export const publishPage = async (
                 createMany: {
                   data: properties
                 }
-              }
+              },
+              blocks: []
             }
           }
         },
@@ -443,7 +448,8 @@ export const publishPage = async (
               createMany: {
                 data: properties
               }
-            }
+            },
+            blocks: []
           },
           update: {
             ...revision,
@@ -454,7 +460,8 @@ export const publishPage = async (
               createMany: {
                 data: properties
               }
-            }
+            },
+            blocks: []
           }
         }
       },
