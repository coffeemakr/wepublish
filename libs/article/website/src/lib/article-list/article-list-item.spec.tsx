import {runStorybookTests} from '@wepublish/testing/website'
import * as stories from './article-list-item.stories'

describe('ArticleList Item', () => {
  runStorybookTests(stories)
})
