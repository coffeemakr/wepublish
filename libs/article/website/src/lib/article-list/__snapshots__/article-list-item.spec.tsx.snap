// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`ArticleList Item should render Default 1`] = `
<DocumentFragment>
  .emotion-0 {
  margin: 0;
  font: inherit;
  color: #1976d2;
  -webkit-text-decoration: underline;
  text-decoration: underline;
  text-decoration-color: rgba(25, 118, 210, 0.4);
  display: grid;
  gap: 16px;
  text-align: left;
  -webkit-text-decoration: none;
  text-decoration: none;
  color: inherit;
  -webkit-align-items: stretch;
  -webkit-box-align: stretch;
  -ms-flex-align: stretch;
  align-items: stretch;
  -webkit-align-content: flex-start;
  -ms-flex-line-pack: flex-start;
  align-content: flex-start;
}

.emotion-0:hover {
  text-decoration-color: inherit;
}

.emotion-1 {
  display: grid;
  width: 100%;
}

.emotion-3 {
  max-width: 100%;
  height: auto;
  aspect-ratio: auto 0.6666666666666666;
  max-height: 80lvh;
  object-fit: contain;
  min-height: inherit;
  object-fit: cover;
}

.emotion-7 {
  margin: 0;
  font-family: "Roboto","Helvetica","Arial",sans-serif;
  font-weight: 400;
  font-size: 1.5rem;
  line-height: 1.334;
  letter-spacing: 0em;
}

.emotion-8 {
  margin-top: 16px;
}

<a
    class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways emotion-0"
    href="https://example.com"
  >
    <div
      class="emotion-1 emotion-2"
    >
      <img
        alt="DSC07717"
        class="emotion-3 emotion-4"
        loading="lazy"
        srcset="https://unsplash.it/800/800 800w,
https://unsplash.it/500/500 500w,
https://unsplash.it/300/300 300w,
https://unsplash.it/200/200 200w"
        title=""
      />
    </div>
    <div
      class="emotion-5 emotion-6"
    >
      <time
        datetime="2023-01-01T00:00:00.000Z"
      >
        01.01.2023
      </time>
      <h1
        class="MuiTypography-root MuiTypography-h5 emotion-7"
      >
        title
      </h1>
      <div
        class="emotion-8 emotion-9"
      >
        Foobar, Foobar
      </div>
    </div>
  </a>
</DocumentFragment>
`;

exports[`ArticleList Item should render WithClassName 1`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -1,8 +1,8 @@
  <DocumentFragment>
    <a
-     class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1vgz1ax-MuiTypography-root-MuiLink-root-ArticleListItem"
+     class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways extra-classname css-1vgz1ax-MuiTypography-root-MuiLink-root-ArticleListItem"
      href="https://example.com"
    >
      <div
        class="css-14c86dg-ArticleListItemImageWrapper e1tp74qn2"
      >"
`;

exports[`ArticleList Item should render WithEmotion 1`] = `
"Snapshot Diff:
Compared values have no visual difference."
`;

exports[`ArticleList Item should render WithoutImage 1`] = `
"Snapshot Diff:
Compared values have no visual difference."
`;

exports[`ArticleList Item should render WithoutImageWithoutImageBlock 1`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -3,22 +3,11 @@
      class="MuiTypography-root MuiTypography-inherit MuiLink-root MuiLink-underlineAlways css-1vgz1ax-MuiTypography-root-MuiLink-root-ArticleListItem"
      href="https://example.com"
    >
      <div
        class="css-14c86dg-ArticleListItemImageWrapper e1tp74qn2"
-     >
-       <img
-         alt="DSC07717"
-         class="css-11fw878-ImageWrapper-ArticleListItem e1shv2yg0"
-         loading="lazy"
-         srcset="https://unsplash.it/800/800 800w,
- https://unsplash.it/500/500 500w,
- https://unsplash.it/300/300 300w,
- https://unsplash.it/200/200 200w"
-         title=""
      />
-     </div>
      <div
        class="css-1w8pg9w-ArticleListItemContent e1tp74qn1"
      >
        <time
          datetime="2023-01-01T00:00:00.000Z""
`;

exports[`ArticleList Item should render WithoutTitle 1`] = `
"Snapshot Diff:
- First value
+ Second value

@@ -23,15 +23,10 @@
        <time
          datetime="2023-01-01T00:00:00.000Z"
        >
          01.01.2023
        </time>
-       <h1
-         class="MuiTypography-root MuiTypography-h5 css-ag7rrr-MuiTypography-root"
-       >
-         title
-       </h1>
        <div
          class="css-1o8pv0h-ArticleListItemAuthors e1tp74qn0"
        >
          Foobar, Foobar
        </div>"
`;
